File: go-client-sdk/pkg/models/shared/eventcreate.go
package shared

type EventCreate struct {
	AccountID     *string                `json:"accountId,omitempty"`
	Additional    map[string]interface{} `json:"additional,omitempty"`
	ContactID     *string                `json:"contactId,omitempty"`
	DealID        *string                `json:"dealId,omitempty"`
	Description   *string                `json:"description,omitempty"`
	EndDateTime   string                 `json:"endDateTime"`
	IsAllDayEvent *bool                  `json:"isAllDayEvent,omitempty"`
	LeadID        *string                `json:"leadId,omitempty"`
	Location      *string                `json:"location,omitempty"`
	OwnerUserID   *string                `json:"ownerUserId,omitempty"`
	StartDateTime string                 `json:"startDateTime"`
	Subject       *string                `json:"subject,omitempty"`
	Type          *string                `json:"type,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/eventcreate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class EventCreate extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=accountId" })
  accountId?: string;

  @SpeakeasyMetadata({ data: "json, name=additional" })
  additional?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=contactId" })
  contactId?: string;

  @SpeakeasyMetadata({ data: "json, name=dealId" })
  dealId?: string;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=endDateTime" })
  endDateTime: string;

  @SpeakeasyMetadata({ data: "json, name=isAllDayEvent" })
  isAllDayEvent?: boolean;

  @SpeakeasyMetadata({ data: "json, name=leadId" })
  leadId?: string;

  @SpeakeasyMetadata({ data: "json, name=location" })
  location?: string;

  @SpeakeasyMetadata({ data: "json, name=ownerUserId" })
  ownerUserId?: string;

  @SpeakeasyMetadata({ data: "json, name=startDateTime" })
  startDateTime: string;

  @SpeakeasyMetadata({ data: "json, name=subject" })
  subject?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;
}
